### PR TITLE
Invite storage

### PR DESCRIPTION
**Pull request by alyakimenko**
- Defines new interface for invite storage.
- Adds invite storage for BoltDB, memory, DynamoDB and MongoDB.
- Adds API endpoints for invites.
    - `GET /invite` - returns all invites with optional `withValid`, `skip` and `limit` parameters.
    - `GET /invite/{id:[a-zA-Z0-9]+}` - returns invite by id.
    - `DELETE /invite/{id:[a-zA-Z0-9]+}` - invalidates (sets `valid` field to false) an invite.
- Updates database composer and server settings.
- Updates all `server-config.yaml` files.

### CODE REVIEW

File: model/invite.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"time"
+)
+
+// Invite is a representation of the invite model.
+// Token field is required for proper working.
+type Invite struct {
+	ID        string    `json:"id" bson:"_id"`
+	AppID     string    `json:"appId" bson:"appId"`
+	Token     string    `json:"token" bson:"token"`
+	Valid     bool      `json:"valid" bson:"valid"`
```
Comment (erudenko) [inline model/invite.go:14]: this one is conflicts with 'Validate', should be something like 'archived' if this is a flag that set explicitly

File: server/boltdb/composer.go [hunk 2]
```diff
@@ -36,40 +38,46 @@ func (dc *DatabaseComposer) Compose() (
...
 	model.TokenBlacklist,
 	model.VerificationCodeStorage,
+	model.InviteStorage,
 	error,
 ) {
 	// We assume that all BoltDB-backed storages share the same filepath, so we can pick any of them.
 	db, err := boltdb.InitDB(dc.settings.Storage.AppStorage.Path)
 	if err != nil {
-		return nil, nil, nil, nil, nil, err
+		return nil, nil, nil, nil, nil, nil, err
```
Comment (erudenko) [inline server/boltdb/composer.go:47]: this thing getting crazier and crazier.

File: web/api/routes.go [hunk 3]
```diff
@@ -46,6 +46,18 @@ func (ar *Router) initRoutes() {
 		ar.Token(model.TokenTypeAccess, nil),
 		negroni.Wrap(ar.RequestInviteLink()),
 	)).Methods("POST")
+	auth.Path(`/{invite:invite/?}`).Handler(negroni.New(
+		ar.Token(model.TokenTypeAccess, nil),
```
Comment (erudenko) [inline web/api/routes.go:50]: does it mean every user has access to the list of invites?

File: web/api/routes.go [hunk 4]
```diff
@@ -46,6 +46,18 @@ func (ar *Router) initRoutes() {
...
+		negroni.Wrap(ar.GetAllInvites()),
+	)).Methods("GET")
+	auth.Path(`/invite/{id:[a-zA-Z0-9]+}`).Handler(negroni.New(
+		ar.Token(model.TokenTypeAccess, nil),
+		negroni.Wrap(ar.GetInviteByID()),
+	)).Methods("GET")
+	auth.Path(`/invite/{id:[a-zA-Z0-9]+}`).Handler(negroni.New(
+		ar.Token(model.TokenTypeAccess, nil),
```
Comment (erudenko) [inline web/api/routes.go:58]: does it mean any user could delete invites?

**Review by erudenko — CHANGES_REQUESTED** [review]

File: web/api/routes.go (hunk 3 above)
Comment (alyakimenko) [reply web/api/routes.go]: Moved to the `admin` router

File: web/api/routes.go (hunk 4 above)
Comment (alyakimenko) [reply web/api/routes.go]: Moved to the `admin` router alongside with the `getInviteByID` endpoint

File: model/invite.go (hunk 1 above)
Comment (alyakimenko) [reply model/invite.go]: Renamed to `Archived`

**Pull request merged**